File: LMS7002M_function_mirror/C/functions_aip.c
#include "caip.h"

#include <stdint.h>
#include <stdio.h>

#include <conio.h>


void *aip_init(void)
{

    const char *connector = "COM7";
    uint8_t nic_addr = 1;
    uint8_t port = 0;
    const char *csv_file = "C:/WorkspaceDDigital/LSM7002MNiosProject/IpdiDriver/ID0000200F_config.csv";

    caip_t *aip = caip_init(connector, nic_addr, port, csv_file);

    aip->reset();

    return (void *)aip;
}

void aip_finish(void *aip)
{
    if (aip != NULL)
    {
        ((caip_t *)aip)->finish();
    }
}

void aip_write(void *aip, uint32_t *parameters, uint32_t nflits, uint32_t offset)
{
    if (aip != NULL)
    {
        ((caip_t *)aip)->writeMem("MProgramMEMIN", parameters, nflits, offset);
    }
}

void aip_start(void *aip)
{
    if (aip != NULL)
    {
        ((caip_t *)aip)->start();
    }
}

uint32_t aip_read(void *aip, uint32_t nflits, uint32_t offset)
{
    uint32_t ctrl_answer[1] = {0};
    if (aip != NULL)
    {
     ((caip_t *)aip)->readMem("MMEMOUT", ctrl_answer, nflits, offset);
    }
    return ctrl_answer[0];
}

void aip_reset(void *aip)
{
    if (aip != NULL)
    {
        ((caip_t *)aip)->reset();
    }
}


void clearDone(void *aip)
{
    if (aip != NULL)
    {
        ((caip_t *)aip)->clearINT(0);
    }
}

uint32_t getStatus(void *aip)
{
    uint32_t STATUS[1];
    if (aip != NULL)
    {
        ((caip_t *)aip)->getStatus(STATUS);
    }
    return STATUS[0];
}


// void enableDone(void *aip)
// {
//     if (aip != NULL)
//     {
//         ((caip_t *)aip)->enableINT(0);
//     }
// }
 

File: LMS7002M_function_mirror/python/functions_AIP.py
from ipdi.ip.pyaip import pyaip, pyaip_init

import sys

def aip_init ():
    try:
        connector = 'COM7'
        nic_addr = 1
        port = 0
        csv_file = 'C:/WorkspaceDDigital/LSM7002MNiosProject/IpdiDriver/ID0000200F_config.csv'

        aip = pyaip_init(connector, nic_addr, port, csv_file)

        aip.reset()

        return aip

    except:
        e = sys.exc_info()
        print('ERROR: ', e)

        aip.finish()
        raise

def aip_finish (aip):
    if aip is not None:
        aip.finish()
   
def aip_write (aip, parameters, nflits, offset):
    if aip is not None:
        aip.writeMem('MProgramMEMIN', parameters, nflits, offset)
    
def aip_start (aip):
    if aip is not None:
        aip.start()

def aip_read (aip, nflits, offset):
    if aip is not None:
        ctrl_answer = aip.readMem('MMEMOUT', nflits, offset)  
        return ctrl_answer
    return None

def aip_reset (aip):
    if aip is not None:
        aip.reset()   

def clearDone (aip):
    if aip is not None:
        aip.clearINT(0)
 

def getStatus (aip):
    if aip is not None:
        STATUS = aip.getStatus()
        return STATUS
    return None
 
    

    
 